### DISCUSSION TITLE
"Show field handles in edit forms" in dev environment only

### DISCUSSION THREAD

**Discussion by turnstylerj** [craftcms/cms#10333] [Ideas]
Currently, the user preference "Show field handles in edit forms" is persistent across environments.

I often find myself disabling this setting in production when demoing a site to a client to clean up the interface to what they'll see, but the next time I pull the database down to my local copy, I need to re-enable it.

Since the typical workflow involves pulling production databases down and overwriting local/dev versions, it'd be nice if this setting had an option to only apply in dev mode (or even a general config like `'showFieldHandles' => $isDev`

**Comment by mmikkel** [comment]
If you want to demo an interface similar to what the client will be seeing, wouldn't it make more sense to impersonate the client's user (or another non-admin account)?

**Reply by turnstylerj** [reply]
Thanks, that's a good idea in the meantime. It still would be nice to have the field handles be something I could enable on an environment level so I don't have to think about it, given showing field handles feels like a feature only really useful in a dev environment.

**Reply by mmikkel** [reply]
Myeah. A config setting for suppressing admin preferences seems a bit redundant to me, when impersonation covers the use case perfectly. But, as long as it's not dependant on `devMode` in any way, sure 😃 

As for the handles, they might not be directly useful in production, but it's _data_ – and for admin users specifically, less is not more. My big pet peeve about `allowAdminChanges` is how [the Settings section is completely hidden away in production](https://github.com/craftcms/cms/issues/4371), making it difficult to quickly review or inspect settings/data/config. The field handles is an exception from that.
